doc/dev/design/FABRID.rst

```diff
@@ -0,0 +1,471 @@
+
+- Author: Justin Rohrer, Jelte van Bommel, Marc Odermatt, Marc Wyss, Cyrill Krähenbühl, Juan A. García-Pardo
+- Last Updated: 2024-04-08
+- Discussion at:
+
+Abstract
+===========
+
+In SCION the endhosts have the option to choose *inter-AS* paths to forward packets to a destination.
+However, some applications require more fine grained path selection like "Do not route traffic over devices
+produced by hardware manufacturer X or that run software Y with version < Z" which requires transparency and
+control also with respect to *intra-AS* paths.
+Such fine-grained path selection is useful for example if there exists a known bug in certain router versions that affect secure communication,
+or if an entity simply does not trust a certain hardware manufacturer, so that traffic can be steered around those devices.
+This can also be seen as an enhancement for keeping traffic within a certain jurisdiction, e.g., by routing traffic
+only along devices located in a specific country.
+
+Background
+===========
+
+The `FABRID <https://netsec.ethz.ch/publications/papers/2023_usenix_fabrid.pdf>`_,
+protocol allows to implement such fine-grained path selection.
+A deployment of FABRID in SCIONLab makes its next-generation features available to interested parties around the globe.
+FABRID also implicitly implements the `EPIC <https://netsec.ethz.ch/publications/papers/Legner_Usenix2020_EPIC.pdf>`_,
+features source authentication for routers and the destination host, and path validation for source and destination hosts.
+
+Proposal
+========
+
+FABRID indroduces policies, which can be thought of as additional path constraints such that the ASes only use intra-AS paths that fulfill that policy.
+The ASes announce those policies to the end hosts, who then can use those additional path constraint during path selection.
+The border routers use those policies to decide on the intra-AS path to forward, e.g. by using MPLS labels.
+Some FABRID policies are globally defined and others locally per AS.
+Global policies make sense in order to have known policies for the common use cases where each end host knows that if an AS supports that policy,
+then it knows the constraints of that policy without the need of fetching the full policy description.
+The AS network operator configures which global FABRID policies are supported for the local AS and can add additional local FABRID
+policies that are valid for this AS.
+The FABRID policy identifiers are provided to the endhosts inside of the PCB or as an detachable PCB extension.
+The FABRID policy descriptions on the other hand have to be requested from the local control service which will then either return
+the cached FABRID policy description or query the policy description from the remote AS.
+That way we can minimize the overhead that the FABRID polices cause to the PCB size.
+A source endhost can then select a path together with FABRID policies and forward the FABRID packet over this path to a destination endhost.
```

Comment:
"select a SCION forwarding path, where it encodes up to one policy index for each on-path AS (it can also select no policy for an AS) in its packets."